### PR TITLE
DynamicDependencies: Only Detour functions in not-packaged processes

### PR DESCRIPTION
**Pull request by DrusTheAxe**
DynamicDependencies only supports not-packaged processes but the Detours hooks were always wired up. Fixed to skip skip that in a packaged process so Microsoft.ProjectReunion.dll can be used in packaged processes for other (not DynDep) features.

Also beefed up parameter validation in the Detour'd GetCurrentPackageInfo[2] to match the OS APIs' parameter validation.

Expanded test cases.

NOTE: Tests verifying package'd process behavior is blocked awaiting [Rewrite DynamicDependencies tests using TAEF #559](https://github.com/microsoft/ProjectReunion/issues/559)

### CODE REVIEW

File: dev/ProjectReunion_DLL/dllmain.cpp [hunk 1]
```diff
@@ -8,8 +8,22 @@
 
 #include <../Detours/detours.h>
 
+static bool IsPackagedProcess()
```
Comment (aeloros) [inline dev/ProjectReunion_DLL/dllmain.cpp:11]: Isn't this like the 4th or 5th copy of this we've made?

File: dev/DynamicDependency/PackageGraphManager.cpp [hunk 2]
```diff
@@ -65,6 +65,12 @@ HRESULT MddCore::PackageGraphManager::GetCurrentPackageInfo2(
     UINT32* count,
     GetCurrentPackageInfo2Function getCurrentPackageInfo2) noexcept try
 {
+    // Check parameter(s) per GetCurrentPackageInfo2 compatibility
+    //   * bufferLength is required
+    //   * buffer is required if bufferLength >0
+    //   * count is required if buffer is not nullptr
+    RETURN_HR_IF(E_INVALIDARG, (bufferLength == nullptr) || (*bufferLength > 0 && buffer == nullptr) || ((count == nullptr) && (buffer != nullptr)));
```
Comment (aeloros) [inline dev/DynamicDependency/PackageGraphManager.cpp:72]: Nit: Horizontal scrollbar when diffing :(  Maybe wrap it?

**Review by aeloros — APPROVED** [review]

File: dev/DynamicDependency/PackageGraphManager.cpp (hunk 2 above)
Comment (DrusTheAxe) [reply dev/DynamicDependency/PackageGraphManager.cpp:72]: The maxlen=100 rule is for *.md (specs), not code.

File: dev/ProjectReunion_DLL/dllmain.cpp (hunk 1 above)
Comment (DrusTheAxe) [reply dev/ProjectReunion_DLL/dllmain.cpp:11]: Lol 2nd or maybe 3rd. I'm planning to centralize this (and a few other things) but waiting on a couple of naming decisions first.

There's a laundry list of refactoring I've got my eye on. Planning to put some time in after I finish some critical path work.

**Pull request merged**